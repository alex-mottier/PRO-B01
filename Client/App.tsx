/**
 * @file    App.tsx
 * @author  Alexis Allemann & Alexandre Mottier
 * @date    22.03.2021
 * @brief   Root component of the application
 */

import * as React from 'react';
import { Provider as PaperProvider } from 'react-native-paper';
import { observer } from 'mobx-react-lite';
import { lightTheme, darkTheme } from './src/app/context/Theme';
import Routes from './src/navigation/Routes';
import * as Font from 'expo-font';
import LoadingComponent from './src/components/Loading/LoadingComponent';
import { useStores } from './src/app/stores/StoresContext';

const App: React.FC = () => {
  /* Usage of MobX global state store */
  const { authenticationStore, themeStore } = useStores();

  /* Component states */
  const [isLoading, setIsLoading] = React.useState(true);

  /**
   * Action when component is loaded
   */
  React.useEffect(() => {
    let mounted = true;
    if (mounted) {
      // Loading icons font
      void Font.loadAsync({
        MaterialCommunityIcons: require('./assets/MaterialCommunityIcons.ttf'),
      }).then(() => {
        setIsLoading(false);
      });
    }
    return () => {
      mounted = false;
    };
  }, []);

  // Don't load the application until the font is loaded
  if (isLoading)
    return (
      <PaperProvider theme={themeStore.theme === 'light' ? lightTheme : darkTheme}>
        <LoadingComponent />
      </PaperProvider>
    );

  return (
    <PaperProvider theme={themeStore.theme === 'light' ? lightTheme : darkTheme}>
      {authenticationStore.isLoading ? <LoadingComponent /> : <Routes />}
    </PaperProvider>
  );
};

export default observer(App);
