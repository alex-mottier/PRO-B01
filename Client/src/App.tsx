/**
 * @file    App.tsx
 * @author  Alexis Allemann
 * @date    22.03.2021
 * @brief   Root component of the application
 */

import * as React from 'react';
import { PreferencesContext } from './app/context/ContextPreferences';
import { Provider as PaperProvider, DefaultTheme, DarkTheme } from 'react-native-paper';
import Globals from './app/context/Globals';
import { RootNavigator } from './navigator/RootNavigator';
import { useColorScheme } from 'react-native';
import { Font } from 'expo';

export default (): React.ReactElement => {
  const colorScheme = useColorScheme();
  const [theme, setTheme] = React.useState<'light' | 'dark'>(
    colorScheme === 'dark' ? 'dark' : 'light',
  );

  function toggleTheme() {
    setTheme((currentTheme) => (currentTheme === 'light' ? 'dark' : 'light'));
  }

  const preferences = React.useMemo(
    () => ({
      toggleTheme,
      theme,
    }),
    [theme],
  );

  return (
    <PreferencesContext.Provider value={preferences}>
      <PaperProvider
        theme={
          theme === 'light'
            ? {
                ...DefaultTheme,
                roundness: 2,
                colors: {
                  ...DefaultTheme.colors,
                  primary: Globals.COLORS.PRIMARY,
                },
              }
            : {
                ...DarkTheme,
                roundness: 2,
                colors: {
                  ...DarkTheme.colors,
                  primary: Globals.COLORS.PRIMARY,
                },
              }
        }>
        <RootNavigator />
      </PaperProvider>
    </PreferencesContext.Provider>
  );
};
function componentWillMount() {
  throw new Error('Function not implemented.');
}
function useFonts(arg0: { Montserrat: any }): [any] {
  throw new Error('Function not implemented.');
}
