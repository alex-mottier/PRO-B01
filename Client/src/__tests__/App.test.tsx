/**
 * @file    App.test.tsx
 * @author  Alexis Allemann
 * @date    22.03.2021
 * @brief   Test class of the main application component
 */

import * as React from 'react';
import * as renderer from 'react-test-renderer';

import App from '../App';

describe('App', () => {
  it('snapshot test', async () => {
    const tree = await renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
