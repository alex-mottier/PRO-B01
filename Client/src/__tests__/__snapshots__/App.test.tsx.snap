// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot test 1`] = `
<View
  style={
    Object {
      "marginTop": 50,
    }
  }
>
  <Text>
    Hello Expo!
  </Text>
</View>
`;
